Slingshot.Api/Concrete/BitbucketRepository.cs

```diff
@@ -302,14 +309,28 @@ public BitbucketRepository(Uri uri, string host, string token)
             }
         }
 
-        public class OAuthInfo
+        /// <summary>
+        /// Partial object, full reponse please reference
+        /// https://confluence.atlassian.com/bitbucket/pullrequests-resource-423626332.html#pullrequestsResource-GETaspecificpullrequest
+        /// </summary>
+        public class BitBucketPullRequest : IPullRequestInfo
+        {
+            public JObject source { get; set; }
+
+            public string SourceBranch
+            {
+                get
+                {
+                    return this.source["branch"]["name"].Value<string>();
```

Comment:
good point, although stacktrace will tell it is from "SourceBranch" property. but since it will show up in the UI, will be good to have a nicely message. added exception handling with a more meaningful message.